src/Model/ApiResourceBase.php

```diff
@@ -153,52 +160,64 @@ public function ensureFull()
     /**
      * Get a resource by its ID.
      *
-     * @param string          $id            The ID of the resource, or the
-     *                                       full URL.
-     * @param string          $collectionUrl The URL of the collection.
-     * @param ClientInterface $client        A suitably configured Guzzle
-     *                                       client.
+     * @param PlatformClient $client  A suitably configured Platform client.
+     * @param string|int     $id      The ID of the resource.
      *
-     * @return static|false The resource object, or false if the resource is
-     *                      not found.
+     * @return static|false The resource object, or false if the resource is not found.
      */
-    public static function get($id, $collectionUrl = null, ClientInterface $client)
+    public static function get(PlatformClient $client, $id)
     {
+        $url = $client->getConnector()->getAccountsEndpoint().static::COLLECTION_PATH;
+
         try {
-            $url = $collectionUrl ? rtrim($collectionUrl, '/') . '/' . urlencode($id) : $id;
-            $request = new Request('get', $url);
-            $data = self::send($request, $client);
+            $data = $client->getConnector()->send(static::COLLECTION_PATH);
 
-            return new static($data, $url, $client, true);
+            // @todo: next level: remove url and guzzle client from the constructor. PlatformClient should be enough.
+            return new static($data, $url, $client->getConnector()->getClient(), true);
         } catch (BadResponseException $e) {
             $response = $e->getResponse();
-            if ($response && $response->getStatusCode() === 404) {
+            // The API may throw either 404 (not found) or 422 (the requested entity id does not exist).
+            if ($response && in_array($response->getStatusCode(), [404, 422])) {
                 return false;
             }
             throw $e;
         }
     }
 
+    /**
+     * Get a collection of resources.
+     *
+     * @param PlatformClient $client  A suitably configured Platform client.
+     * @param QueryInterface $query   An instance of query interface. It will be used to build a guzzle query.
+     *
+     * @return Collection;
+     */
+    public static function getCollection(PlatformClient $client, ?QueryInterface $query = null)
```

Comment:
Presumably it doesn't need the `= null` if it already has a nullable type hint

Reply:
Unfortunately, it does need it. It returns a "Too few arguments" error unless `null` explicitly provided.